File: src/main/java/banksoft/model/Usuario.java
package banksoft.model;

import jakarta.persistence.*;

@Entity
@Table(name = "usuario")
public class Usuario {
    private Integer idUsuario;

    private Integer idCliente;

    private String nombreUsuario;

    private String contrasena;

    private String estatus;
 
    private String correo;

    public Usuario() {}

    public Usuario(String nombreUsuario, String contrasena) {
        this.nombreUsuario = nombreUsuario;
        this.contrasena = contrasena;
        this.estatus = "A";
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usuario")
    public Integer getIdUsuario() { return idUsuario; }
    public void setIdUsuario(Integer idUsuario) { this.idUsuario = idUsuario; }

    @Column(name = "id_cliente")
    public Integer getIdCliente() { return idCliente; }
    public void setIdCliente(Integer idCliente) { this.idCliente = idCliente; }

    @Column(name = "nombre_usuario", length = 30)
    public String getNombreUsuario() { return nombreUsuario; }
    public void setNombreUsuario(String nombreUsuario) { this.nombreUsuario = nombreUsuario; }

    @Column(name = "contrasena", length = 200)
    public String getContrasena() { return contrasena; }
    public void setContrasena(String contrasena) { this.contrasena = contrasena; }

    @Column(name = "estatus", length = 1)
    public String getEstatus() { return estatus; }
    public void setEstatus(String estatus) { this.estatus = estatus; }
 
   /*  @Column(name = "correo", length = 100)
    public String getCorreo() { return correo; }
    public void setCorreo(String correo) { this.correo = correo; }
    */
}

File: transferencia-service/models/usuario.py
from sqlalchemy import Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from utils.database import Base


class Usuario(Base):
    __tablename__ = "usuario"

    id_usuario: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    id_cliente: Mapped[int | None] = mapped_column(Integer, nullable=True)
    nombre_usuario: Mapped[str | None] = mapped_column(String(30), nullable=True)
    contrasena: Mapped[str | None] = mapped_column(String(200), nullable=True)
    estatus: Mapped[str | None] = mapped_column(String(1), nullable=True)
    correo: Mapped[str | None] = mapped_column(String(200), nullable=True)